tool/tsh/proxy.go

```diff
@@ -442,6 +442,7 @@ func onProxyCommandDB(cf *CLIConf) error {
 			dbcmd.WithNoTLS(),
 			dbcmd.WithLogger(log),
 			dbcmd.WithPrintFormat(),
+			dbcmd.WithTolerateMissingCLIClient(),
```

Comment:
This PR made me take another look at `tolerateMissingCLIClient`. I added this back in the day but now that I understand this part of the codebase better, I think in the future we should be able to remove this option in favor of just using `noTLS`.

From what I see in `getMySQLCommand`, the only reason we need the mysql bin to be present is to decide whether the db is of MariaDB flavor in the absence of the mariadb bin.

However, even if the mysql bin is present, there's no difference in args passed to the actual command if `noTLS` is present:

https://github.com/gravitational/teleport/blob/ae90d5f4dfb11652d8c7ec69a528ddc6265baa88/lib/client/db/dbcmd/dbcmd.go#L290-L295

https://github.com/gravitational/teleport/blob/ae90d5f4dfb11652d8c7ec69a528ddc6265baa88/lib/client/db/dbcmd/dbcmd.go#L314-L319

`getMySQLCommand` should have a separate branch for `options.noTLS` at the start of the method. This branch would pick mariadb over mysql bin if present and then pass `getMySQLCommonCmdOpts` to `c.options.exe.Command`. Then `getMariaDBArgs` and `getMySQLOracleCommand` would not have to worry about `noTLS` at all.

---

The only situation where we want the code to fail if both binaries are missing is `tsh db connect`. But the code which handles this command will never pass `noTLS` so `noTLS` can become the equivalent of `tolerateMissingCLIClient`.

Reply:
Ah, thanks, I missed that this can happen.